### PR TITLE
feat(plp): product detail quick view from plp

### PR DESCRIPTION
**Pull request by kibo-geetanshuchhabra**

### CODE REVIEW

File: components/page-templates/ProductDetail/ProductDetailTemplate.tsx [hunk 1]
```diff
@@ -41,12 +43,25 @@ import type {
 
 interface ProductDetailTemplateProps {
   product: ProductCustom
-  breadcrumbs: BreadCrumb[]
-  cmsProducts: any
+  breadcrumbs?: BreadCrumb[]
+  cmsProducts?: any
```
Comment (kibo-sushant) [inline components/page-templates/ProductDetail/ProductDetailTemplate.tsx:47]: use specific type

File: components/page-templates/ProductDetail/ProductDetailTemplate.tsx [hunk 2]
```diff
@@ -372,18 +397,24 @@ const ProductDetailTemplate = (props: ProductDetailTemplateProps) => {
...
-            <Box paddingY={3}>
-              <ProductInformation productFullDescription={description} options={properties} />
-            </Box>
-          )}
-        </Grid>
-
-        {cmsProducts?.components?.length > 0 &&
+        {!isQuickViewModal && (
+          <>
+            {' '}
```
Comment (kibo-sushant) [inline components/page-templates/ProductDetail/ProductDetailTemplate.tsx:402]: remove unnecessary code

File: components/page-templates/ProductDetail/ProductDetailTemplate.tsx [hunk 3]
```diff
@@ -372,18 +397,24 @@ const ProductDetailTemplate = (props: ProductDetailTemplateProps) => {
...
+              {properties?.length > 0 && (
+                <Box paddingY={3}>
+                  <ProductInformation productFullDescription={description} options={properties} />
+                </Box>
+              )}
+            </Grid>
+          </>
+        )}
+        ​
+        {!isQuickViewModal &&
```
Comment (kibo-sushant) [inline components/page-templates/ProductDetail/ProductDetailTemplate.tsx:416]: Why are we repeating condition {!isQuickViewModal

File: components/product/ProductCard/ProductCard.tsx [hunk 4]
```diff
@@ -36,8 +39,12 @@ export interface ProductCardProps {
   isInCart?: boolean
   isLoading?: boolean
   isShopNow?: boolean
+  isShown?: boolean
   isShowWishlistIcon?: boolean
+  product?: Product
+  isQuickViewModal?: boolean
   onAddOrRemoveWishlistItem?: () => void
+  fromProductListingPage?: boolean
```
Comment (kibo-sushant) [inline components/product/ProductCard/ProductCard.tsx:47]: No mandatory props? How all the Props can be nullable?

File: components/product/ProductCard/ProductCard.tsx [hunk 5]
```diff
@@ -68,6 +75,8 @@ const ProductCardSkeleton = () => {
 }
 
 const ProductCard = (props: ProductCardProps) => {
+  //setting state for displaying quickview button
+  const [isQuickViewShown, setIsQuickViewShown] = useState(false)
```
Comment (kibo-sushant) [inline components/product/ProductCard/ProductCard.tsx:79]: remove commented code and use Type explicitly

File: components/product/ProductQuickViewModal/ProductQuickViewDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+const ProductQuickViewDialog = (props: ProductQuickViewDialogProps) => {
+  const { closeModal } = useModalContext()
+  return (
+    <KiboDialog
+      showCloseButton
+      showContentTopDivider={false}
+      showContentBottomDivider={false}
+      Actions={''}
+      Content={
+        <ProductDetailTemplate product={props.product} isQuickViewModal={props.isQuickViewModal} />
```
Comment (kibo-sushant) [inline components/product/ProductQuickViewModal/ProductQuickViewDialog.tsx:20]: destructure props and use it

File: components/product/ProductQuickViewModal/ProductQuickViewDialog.tsx [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+
+import KiboDialog from '@/components/common/KiboDialog/KiboDialog'
+import { ProductDetailTemplate } from '@/components/page-templates'
+import { useModalContext } from '@/context/ModalContext'
+import type { ProductCustom } from '@/lib/types'
+interface ProductQuickViewDialogProps {
```
Comment (kibo-sushant) [inline components/product/ProductQuickViewModal/ProductQuickViewDialog.tsx:7]: for readability purposes separate various groups with the space.
Ex. provide space before and after the interface

File: components/product/ProductQuickViewModal/ProductQuickViewDialog.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+  title: 'Product/ProductQuickViewDialog',
+} as ComponentMeta<typeof ProductQuickViewDialog>
+const Template: ComponentStory<typeof ProductQuickViewDialog> = (args) => (
+  <ProductQuickViewDialog {...args} />
+)
+export const Common = Template.bind({})
+Common.args = {
+  product: ProductCustomMock,
+  isQuickViewModal: true,
+}
```
Comment (kibo-sushant) [inline components/product/ProductQuickViewModal/ProductQuickViewDialog.stories.tsx:19]: for readability purposes separate various groups with the space.

File: components/product/ProductCard/ProductCard.tsx [hunk 9]
```diff
@@ -82,67 +91,93 @@ const ProductCard = (props: ProductCardProps) => {
     isShopNow = false,
     isInWishlist = false,
     isShowWishlistIcon = true,
+    isQuickViewModal = true,
+    product,
     onAddOrRemoveWishlistItem,
+    fromProductListingPage,
```
Comment (amolpawar-kibo) [inline components/product/ProductCard/ProductCard.tsx:97]: Assign default value for fromProductListingPage

File: components/common/KiboDialog/KiboDialog.tsx [hunk 10]
```diff
@@ -110,18 +110,18 @@ const KiboDialog = (props: KiboDialogProps) => {
       data-test-id="kibo-dialog"
     >
       <Container maxWidth={'xl'}>
-        {Title && (
-          <StyledDialogTitle id="kibo-dialog-title">
+        <StyledDialogTitle id="kibo-dialog-title">
+          {Title && (
```
Comment (amolpawar-kibo) [inline components/common/KiboDialog/KiboDialog.tsx:114]: Is there a closed button with no title?

File: components/page-templates/ProductDetail/ProductDetailTemplate.tsx (hunk 1 above)
Comment (kibo-nehabhanushali) [reply components/page-templates/ProductDetail/ProductDetailTemplate.tsx:47]: @kibo-sushant This is going to be removed once your PR is merged

File: components/product/ProductCard/ProductCard.tsx (hunk 9 above)
Comment (kibo-geetanshuchhabra) [reply components/product/ProductCard/ProductCard.tsx:97]: Resolved

File: components/common/KiboDialog/KiboDialog.tsx (hunk 10 above)
Comment (kibo-geetanshuchhabra) [reply components/common/KiboDialog/KiboDialog.tsx:114]: yes(We separate out title and close button).

File: components/product/ProductQuickViewModal/ProductQuickViewDialog.tsx (hunk 7 above)
Comment (kibo-geetanshuchhabra) [reply components/product/ProductQuickViewModal/ProductQuickViewDialog.tsx:7]: Resolved

File: components/product/ProductQuickViewModal/ProductQuickViewDialog.tsx (hunk 6 above)
Comment (kibo-geetanshuchhabra) [reply components/product/ProductQuickViewModal/ProductQuickViewDialog.tsx:20]: Resolved

File: components/page-templates/ProductDetail/ProductDetailTemplate.tsx (hunk 3 above)
Comment (kibo-geetanshuchhabra) [reply components/page-templates/ProductDetail/ProductDetailTemplate.tsx:416]: Resolved

File: components/page-templates/ProductDetail/ProductDetailTemplate.tsx (hunk 2 above)
Comment (kibo-geetanshuchhabra) [reply components/page-templates/ProductDetail/ProductDetailTemplate.tsx:402]: Resolved

File: components/product/ProductCard/ProductCard.tsx [hunk 11]
```diff
@@ -36,8 +37,13 @@ export interface ProductCardProps {
   isInCart?: boolean
   isLoading?: boolean
   isShopNow?: boolean
+  isShown?: boolean
```
Comment (kibo-chandradeeptalaha) [inline components/product/ProductCard/ProductCard.tsx:40]: remove this prop if not used.

File: components/product/ProductCard/ProductCard.tsx [hunk 12]
```diff
@@ -36,8 +37,13 @@ export interface ProductCardProps {
...
   isShowWishlistIcon?: boolean
+  product?: Product
+  isQuickViewModal?: boolean
```
Comment (kibo-chandradeeptalaha) [inline components/product/ProductCard/ProductCard.tsx:43]: remove this prop if not used

File: components/product/ProductCard/ProductCard.tsx [hunk 13]
```diff
@@ -83,66 +89,87 @@ const ProductCard = (props: ProductCardProps) => {
...
+                    size="small"
+                    icon={<StarRounded color="primary" data-testid="filled-rating" />}
+                    emptyIcon={<StarRounded data-testid="empty-rating" />}
+                    data-testid="product-rating"
                   />
+                  {fromProductListingPage && (
+                    <Button
+                      variant="contained"
+                      color="secondary"
+                      sx={{ width: '100%', marginTop: '1 rem', opacity: isQuickViewShown ? 1 : 0 }}
```
Comment (kibo-chandradeeptalaha) [inline components/product/ProductCard/ProductCard.tsx:164]: Can we use css hover for opacity change instead of creating a state ?

**Review by kibo-chandradeeptalaha — APPROVED** [review]

**Review by kibo-sushant — APPROVED** [review]

**Review by kibo-kevinwatts — APPROVED** [review]

**Pull request merged**